cranelift/codegen/src/settings.rs

```diff
@@ -115,6 +115,24 @@ impl Value {
     }
 }
 
+impl fmt::Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Some(enum_variant) = self.as_enum() {
```

Comment:
Can we write this if-chain as a `match` on `self.detail`?

Reply:
`self.details` doesn't directly contain the value. It only contains information how to get the value. The logic is non-trivial for some types, so I wanted to avoid duplication.

Reply:
Fair enough, this is reasonable too.